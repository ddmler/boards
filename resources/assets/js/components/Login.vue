<template>
  <div>
    <div 
      v-if="error" 
      class="notification is-danger">
      <p>It looks like those credentials are not working.</p>
    </div>
    <form 
      autocomplete="off" 
      method="post" 
      @submit.prevent="login">
      <div class="field">
        <label 
          for="email" 
          class="label">E-mail</label>
        <input 
          id="email" 
          v-model="email" 
          type="email" 
          placeholder="user@example.com" 
          class="input" 
          required>
      </div>
      <div class="field">
        <label 
          for="password" 
          class="label">Password</label>
        <input 
          id="password" 
          v-model="password" 
          type="password" 
          class="input" 
          required>
      </div>
      <button 
        type="submit" 
        class="button">Sign in</button>
    </form>
  </div>
</template>
<script>
  export default {
    data(){
      return {
        email: null,
        password: null,
        error: false
      }
    },
    methods: {
      login(){
        var app = this
        this.$auth.login({
            params: {
              email: app.email,
              password: app.password
            },
            success: function () {},
            error: function () { this.error = true; },
            rememberMe: true,
            redirect: '/dashboard',
            fetchUser: true,
        });
      },
    }
  }
</script>
