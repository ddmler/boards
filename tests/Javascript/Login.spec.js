import { mount } from '@vue/test-utils';
import Login from '@/Login.vue';

describe('Login.vue', () => {

    it('matches the snapshot', () => {
        const wrapper = mount(Login)
        expect(wrapper.html()).toMatchSnapshot();
    });

});