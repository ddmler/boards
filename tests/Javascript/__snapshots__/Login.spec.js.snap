// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue matches the snapshot 1`] = `
<div>
  <form autocomplete="off" method="post">
    <div class="field">
      <label for="email" class="label">E-mail</label>
      <input id="email" type="email" placeholder="user@example.com" required="required" class="input">
    </div>
    <div class="field">
      <label for="password" class="label">Password</label>
      <input id="password" type="password" required="required" class="input">
    </div>
    <button type="submit" class="button">Sign in</button>
  </form>
</div>
`;
